pkg/agent/protocol/protocol.go

```diff
@@ -89,7 +90,7 @@ func NewDisruptor(
 }
 
 // Apply applies the Disruption to the target system
-func (d *disruptor) Apply(duration time.Duration) error {
+func (d *disruptor) Apply(ctx context.Context, duration time.Duration) error {
```

Comment:
I see that we're passing down both a context and a duration. What happened to the idea of using the context deadline as the duration?